src/main/java/frc/robot/commands/ArmCommandSelector.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import java.util.function.Supplier;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.config.ArmConfig.ArmPosition;
+import frc.robot.config.ArmConfig.ArmSetpoint;
+import frc.robot.robotcontainers.CompRobotContainer.RobotGamePieceState;
+
+public class ArmCommandSelector extends CommandBase {
+  RobotGamePieceState m_robotState;
+  ArmPosition m_Position;
+  ArmSetpoint m_ArmSetPoint;
+  boolean m_slowerAcceleration;
+  ArmCommand m_ArmCommand;
+  
+  /** Creates a new ArmCommand. */
+  public ArmCommandSelector(Supplier<RobotGamePieceState> robotState, 
+                            ArmPosition position,
+                            boolean slowerAcceleration ) {
+    //get the current robotState: from container
+    m_robotState = robotState.get();
```

Comment:
The robotState will change a lot during a match. Here you're getting the robot state when the command is first constructed and then only using that state. Instead, you will need to get the state in initialize() and then check which state it is.